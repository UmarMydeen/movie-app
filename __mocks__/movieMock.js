const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;