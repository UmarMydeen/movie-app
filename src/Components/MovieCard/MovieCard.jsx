import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link, BrowserRouter } from 'react-router-dom';

const MovieCard = ({
    movie: {
        imdbID,
        Poster: imageUrl,
        Title: title,
    }
}) => {
    return (
        <BrowserRouter>
            <Link to={`/movieDetails/${imdbID}`}>
                <div data-testid={imdbID} className="card m-1" key={imdbID}>
                    <img src={imageUrl} className="card-img-top" alt="..." />
                    <div className="card-body">
                        <h5 className="card-title">{title}</h5>
                    </div>
                </div>
            </Link >
        </BrowserRouter>
    );
}

MovieCard.propTypes = {
    movie: PropTypes.shape({
        imdbID: PropTypes.string.isRequired,
        Poster: PropTypes.string.isRequired,
        Title: PropTypes.string.isRequired,
    })
};

export default MovieCard;