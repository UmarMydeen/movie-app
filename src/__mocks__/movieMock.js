const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;