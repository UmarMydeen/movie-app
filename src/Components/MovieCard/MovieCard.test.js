import { getByTestId, render, screen } from '@testing-library/react';
import MovieCard from './MovieCard';
import movieMock from '../../../__mocks__/movieMock';

let mockProps;

beforeEach(() => {
    mockProps = {
        movie: movieMock
    }
})

test('should render component correctly', () => {
    const component = render(<MovieCard {...mockProps} />);
    expect(component).toMatchSnapshot();
});