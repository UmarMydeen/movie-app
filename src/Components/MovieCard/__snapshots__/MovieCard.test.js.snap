// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/movieDetails/mockimdbID1"
      >
        <div
          class="card m-1"
          data-testid="mockimdbID1"
        >
          <img
            alt="..."
            class="card-img-top"
            src="mockPoster"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              mockTitle
            </h5>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/movieDetails/mockimdbID1"
    >
      <div
        class="card m-1"
        data-testid="mockimdbID1"
      >
        <img
          alt="..."
          class="card-img-top"
          src="mockPoster"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            mockTitle
          </h5>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
